### PR TITLE
Refactor AutoSave and add tests (and fix misc bugs)

### PR DESCRIPTION
**Pull request by richrace**
* Extract the function excludeSavedAmendments (with unit tests)
* Fix sticky Exceptions/Trigger filter chip (flakey cypress e2e test)
* Add cypress component test for AutoSave component
* Add cypress e2e tests for updating notes - proves the court case got updated
* Fix submit button ignores amendments already saved and add e2e cypress tests

### CODE REVIEW

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-asn.cy.ts [hunk 1]
```diff
@@ -286,4 +329,47 @@ describe("ASN", () => {
   it("Should not display an editable field for ASN when exceptionsEnabled is false for user", () => {
     loginAndVisit("NoExceptionsFeatureFlag", "/bichard/court-cases/0")
   })
+
+  describe("when I submit resolved exceptions I should not the same value in the notes", () => {
```
Comment (C-gyorfi) [inline cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-asn.cy.ts:333]:
```suggestion
  describe("when I submit resolved exceptions I should not see the same value in the notes", () => {
```

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts [hunk 2]
```diff
@@ -404,4 +444,50 @@ describe("NextHearingLocation", () => {
     cy.get("#next-hearing-location-row .warning-icon").should("exist")
     cy.contains("Select valid Next hearing location").should("exist")
   })
+
+  describe("when I submit resolved exceptions I should not the same value in the notes", () => {
```
Comment (C-gyorfi) [inline cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts:448]:
```suggestion
  describe("when I submit resolved exceptions I should not see the same value in the notes", () => {
```

File: src/features/CourtCaseDetails/Sidebar/Exceptions.tsx [hunk 3]
```diff
@@ -46,6 +48,12 @@ const Exceptions = ({ onNavigate, canResolveAndSubmit, stopLeavingFn }: Props) =
     stopLeavingFn(false)
   }
 
+  const amendmentsToUpdate = excludeSavedAmendments(
```
Comment (C-gyorfi) [inline src/features/CourtCaseDetails/Sidebar/Exceptions.tsx:51]: 🎉

File: src/utils/autoSave/excludeSavedAmendments.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }
+  })
+
+  if (!isEmpty(newValues)) {
+    map.set(amendmentField, newValues)
+  }
+}
+
+const excludeSavedAmendments = (
+  amendmentFields: AmendmentKeys[],
```
Comment (C-gyorfi) [inline src/utils/autoSave/excludeSavedAmendments.ts:35]: I wonder if it's necessary to pass the amendmentFields as an argument

File: src/components/EditableFields/AutoSave.tsx [hunk 5]
```diff
@@ -104,3 +96,5 @@ export const AutoSave = ({
     </>
   )
 }
+
+export default AutoSave
```
Comment (C-gyorfi) [inline src/components/EditableFields/AutoSave.tsx:100]: Is export default the preferred way of exporting a function/component? I liked that export const would ensure we use the same name at import.

**Review by C-gyorfi — DISMISSED** [review]
Awesome work, added some comments but nothing that would block the merge 👏

File: src/components/EditableFields/AutoSave.tsx (hunk 5 above)
Comment (richrace) [reply src/components/EditableFields/AutoSave.tsx:100]: I prefer exporting the component. Maybe an ADR conversion or best practice? Decide as a team how we manage exporting React Components. There's nothing to stop using the export const function and having the default as well.

File: src/utils/autoSave/excludeSavedAmendments.ts (hunk 4 above)
Comment (richrace) [reply src/utils/autoSave/excludeSavedAmendments.ts:35]: From the AutoSave point of view, it only wants to save a single value, e.g. `asn`

And from the `Submit Exceptions`, it removes the `offenceReasonSequence`, which gets added to the notes.

**Review by C-gyorfi — APPROVED** [review]

**Pull request merged**